staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -140,7 +141,7 @@ type latencyAdapter struct {
 }
 
 func (l *latencyAdapter) Observe(ctx context.Context, verb string, u url.URL, latency time.Duration) {
-	l.m.WithContext(ctx).WithLabelValues(verb, u.String()).Observe(latency.Seconds())
+	l.m.WithContext(ctx).WithLabelValues(verb, u.Host, u.Path).Observe(latency.Seconds())
```

Comment:
we have to remember to use the template thing for the path or we'll have an unbounded var here with a potential huge number of values

Reply:
ah, good point, maybe we could even enforce that here wdyt?

Reply:
nvm it requires a request object so we won't be able to improve it here, we just have to make sure that we record the templated path

Reply:
this is going to be a breaking change...

Reply:
yes, but considering that the metric is only in ALPHA stage and that the label has an unbound cardinality that has been proven to explode, I think it is fine as long as it is documented in the CHANGELOG

Reply:
Stating that things are in ALPHA isn't actually a reason, since that was the default. The criteria for determining whether to be cautious should be (1) is this actually likely to be used by people (2) how badly are we going to break them.

Reply:
Given the nature of the issue (i.e. unbounded cardinality), I actually don't disagree with you. I'm just saying we can't make the ALPHA claim as a reason for breaking something.

Reply:
yes, I totally agree with that, I was just mentioning it to reinforce my point that cardinality explosion is critical enough to modify the metric even though that would break users which is something that could have been more complex if the metric was stable